goth_tests/README.md

```diff
@@ -0,0 +1,267 @@
+# goth integration tests
+
+## Introduction
+
+This is the base directory for all of `yagna`'s integration tests which make use of [`goth`](https://github.com/golemfactory/goth) (GOlem Test Harness).
+
+`goth` is a Python package which is a backend and a runner for `yagna` integration tests. This directory is home to a Python project which defines a number of test cases and makes use of `goth` to execute them on an isolated Golem network.
+
+### Test directory structure
+Tests are organised into a directory structure based on their characteristics:
+```
+.
+├── assets                                  # Common, generated assets for tests
+│   ├── docker
+│   │   ├── yagna-goth.Dockerfile
+│   │   ├── docker-compose.yml              # Compose file for the static part of each test's network
+│   │   └── ...
+│   ├── goth-config.yml                     # Default test network configuration
+│   └── ...
+├── domain                                  # Domain-specific tests
+│   ├── exe_units
+│   ├── market
+│   ├── payments
+│   │   ├── test_zero_amount_txs.py
+│   │   └── ...
+│   └── ya-provider
+│       └── test_provider_multi_activity.py
+├── e2e                                     # End-to-end tests
+│   ├── vm
+│   │   ├── assets                          # Blender-specific assets
+│   │   │   ├── params.json
+│   │   │   ├── scene.blend
+│   │   │   └── ...
+│   │   └── test_e2e_vm.py
+│   └── wasi
+│       └── test_e2e_wasi.py
+```
+In the above structure, for each test there is a separate `test_*.py` file with a single `test_*` function defined inside.
+
+Domain-specific tests are placed in their appropriate directory under `domain`.
+
+If a test requires custom assets (e.g. VM Blender test) they should be placed in a directory named `assets` alongside the test `.py` file itself.
+
+### How these tests work
+Every `goth` test operates on an isolated, local network of `yagna` nodes which gets created using Docker. Besides the `yagna` containers themselves this network also includes components such as `ya-sb-router` or `ganache` local blockchain.
+
+For every test case the following steps are performed:
+1. `docker-compose` is used to start the so-called "static" containers (e.g. local blockchain, these are defined in `docker-compose.yml`) and create a common Docker network for all containers participating in the test.
+2. The test runner creates a number of Yagna containers (as defined in `goth-config.yml`) which are connected to the `docker-compose` network.
+3. For each Yagna container started a so-called "probe" object is created and made available inside the test via the `Runner` object.
+4. The integration test scenario is executed as defined in the function called by `pytest`.
+5. Once the test is finished, all previously started Docker containers (both "static" and "dynamic") are removed.
+
```

Comment:
Maybe it's worth mentioning here where the test logs are stored by default?

Reply:
Good catch, added a section on logs in 31cfad0